### ISSUE TITLE
is this project officially dead ?

### ISSUE THREAD

**Issue by avichou** [transmission/transmission#1840]
just curious as whether we should expect future releases ,such as an ARM build for apple silicon macs for example?

there hasnt been any posts for over a year now

**Comment by XenonPK** [comment]
I'm pretty sure people have been using transmission in raspberry pis and low power ARM based NAS devices for decades now.
I don't think ARM support is a problem for the project.

As far as activity goes, the latest commit has less than 24h of age, so it seems pretty active to me. This does not look like an issue at all...

**Comment by Pentaphon** [comment]
It's pretty obvious from the commits that we are going to have a big release soon. This project needs more attention and management. Github has improved the overall engagement but Transmission is just not getting the same help and hype that qBittorrent is getting. We just have to keep spreading the word about Transmission, especially once they get a stable build out for Windows, and hope that people come to the project.

**Comment by avichou** [comment]
> I'm pretty sure people have been using transmission in raspberry pis and low power ARM based NAS devices for decades now.
> I don't think ARM support is a problem for the project.
> 
> As far as activity goes, the latest commit has less than 24h of age, so it seems active to me. This does not look like an issue at all...

im pretty dumb but i didnt know where to post a comment/discussion

**Comment by Pentaphon** [comment]
> im pretty dumb but i didnt know where to post a comment/discussion

The Transmission site has a forum.

**Comment by stefanos82** [comment]
This list of [merged and approved PRs](https://github.com/transmission/transmission/pulls?q=is%3Apr+is%3Aclosed) speaks volumes about project's activity, therefore it's far from being dead.

It just goes through a phase of transformation from C to modern C++.

In my humble opinion, things will become smoother for candidates who want to offer their contributions to Transmission.

**Comment by Pentaphon** [comment]
> In my humble opinion, things will become smoother for candidates who want to offer their contributions to Transmission.

Yep. This was vital to the future of the project. Transmission is now on par with all the other clients. We might even see people come from other projects now that C++ is being used.

**Comment by xavery** [comment]
> It just goes through a phase of transformation from C to modern C++.

Making C code compile under a C++ compiler (which is precisely what the recent commits do) is light years away from "transforming it to modern C++".

Actually porting a code base like Transmission's to "modern C++" isn't something that can be done in a few commits. Hell, I'd even argue that it isn't needed.

**Comment by xavery** [comment]
See #701 which might be outdated but presents a very valuable perspective.

**Comment by stefanos82** [comment]
> > It just goes through a phase of transformation from C to modern C++.
>
> Making C code compile under a C++ compiler (which is precisely what the recent commits do) is light years away from "transforming it to modern C++".
>
> Actually porting a code base like Transmission's to "modern C++" isn't something that can be done in a few commits. Hell, I'd even argue that it isn't needed.

Seems like you first sent this issue and then decided to delete it; I have received the email though, therefore I will answer on this reply of yours.

In my humble opinion this is *exactly* the reason that *could* attract more candidates; what you can write in modern C++ would take ages in C code, especially around algorithms and standard libraries.

I will repeat myself again the same way I have answered elsewhere; not many people know how to use C **the right way**, let alone safely, thus their hesitant to jump in and start contributing.

It's not a shame for us all to admit that writing C makes it easier to write unsafe code, thus the countless vulnerabilities that exist out there; it takes years of practice to master your craft and learn to be disciplined and write as secure code as possible.

In C++ though you have more mechanisms to use to produce a safer result than C and you have a number of standard libraries that have been battle-tested for decades.

You might not like C++ and I understand your concerns, but it's a great tool with unlimited power and extremely useful toolset that can make your life easier without being forced to reinvent the wheel when you have to deal with libraries.

To clarify one thing though: this is not a "C++ is better than C" language war; all I'm saying is use the right tool for the right job and being honest with you, I firmly believe the upgrade to C++ would reveal many code restrictions Transmission had for years that went unnoticed; they were well-hidden and went undetected, but the migration brought them in light.

**Comment by xavery** [comment]
> Seems like you first sent this issue and then decided to delete it; I have received the email though, therefore I will answer on this reply of yours.
This part of my reply was deleted for a reason : it's not related to the topic that this issue addresses, or tries to address.

What I wanted to say (although poorly, I have to admit) is that it seems weird that the maintainers chose to go ahead with this quite serious and large C++ "switch" while there are (at the time of writing) 763 open issues and 116 open PRs. How many of those PRs contain valueable code? How many will have merge conflicts after all the C++ related changes are merged? Will the original authors care enough to update their branches and resolve these conflicts, or maybe will the maintainers invest their time to resolve these conflicts themselves if the given PR is deemed worthwhile enough?

Of course, since this is a volunteer-based project, I am not (nor is anyone) in any position to require any kind of specific behaviour from the maintainers. I just wanted to address what seems like a surprising choice for a project that's been for a long time called  "fairly feature-complete" (#701). Perhaps there are reasons behind it that I'm not aware of - however, if the main reason is trying to attract new people, there's nothing left for me but to wish all you best of luck.

Just to clarify one last time : I am not in any way opposed to the language itself, otherwise I wouldn't write code in it myself. It's just the timing and the circumstances of this move which are surprising.

**Comment by Pentaphon** [comment]
@xavery it's pretty clear that the maintainers have a specific direction in mind for this project. PRs are not usually taken into consideration, probably because of how long it took for them to switch to C++ and how they want the client to function. Now that they're doing it, future and possibly present PRs can be merged. As long as there's still activity on this fairly mature project, I can't really complain, especially when there's so much work to be done for Windows users, who will bring in more users to the project than ever before.

**Comment by ckerr** [comment]
> while there are (at the time of writing) 763 open issues and 116 open PRs.

Four-month update: 594 open issues, 69 open PRs.

**Comment by Pentaphon** [comment]
@ckerr This issue is closed so I might as well use it to say thank you so much for spending your free time keeping this client modernized, catching up on PRs and making it easier for other people with C++ skills to keep the project going. So many people are obsessed with using uTorrent or qBittorrent so it's great that you and the other contributors are making sure there's another actively maintained open source torrent client around after 16 years. I hope whoever else is reading this is inspired to spend their time, funds or skills to help you with this project or at the very least, tell you that they appreciate you not giving up on this project. Looking forward to the next release!

**Comment by avichou** [comment]
> > while there are (at the time of writing) 763 open issues and 116 open PRs.
> 
> Four-month update: 594 open issues, 69 open PRs.
> 
> Also [202 of those open issue are "pr welcome"](https://github.com/transmission/transmission/issues?q=is%3Aissue+is%3Aopen+label%3A%22pr+welcome%22+) so they are triaged, accepted in principle, and are in limbo until a contributor submits a PR

well indeed but still no stable build since 2020

**Comment by GScottElshaw** [comment]
Have you tried a nightly? You may like it. https://build.transmissionbt.com/ After trying it, you could help make it better.

**Comment by ckerr** [comment]
> well indeed but still no stable build since 2020

Don't be an ass. The project has gone from moribund to full-speed-ahead, but it doesn't make sense to do a release in the middle of a refactor. @GScottElshaw's suggestion to try [the nightly builds](https://transmissionbt.com/download) sounds like a good compromise.

>> while there are (at the time of writing) 763 open issues and 116 open PRs.
>
> Four-month update: 594 open issues, 69 open PRs.

560 open issues, 46 open PRs.

**Comment by Pentaphon** [comment]
> well indeed but still no stable build since 2020

If you want to help that process move along faster, try submitting bug reports and letting people know that this project needs help wherever you can. It's understandable when you want a project you like to keep getting updated but there's positive ways to deal with it.

**Comment by ckerr** [comment]
> > > while there are (at the time of writing) 763 open issues and 116 open PRs.
> > 
> > Four-month update: 594 open issues, 69 open PRs.
> 
> 560 open issues, 46 open PRs.

421 open issues, 34 open PRs.

There is a point where the issue numbers won't shrink much more -- for example, a lot of the issues are enhancement requests marked 'pr welcome' meaning if someone else does the work that's great, but it's not a high priority so the maintainers probably won't do it.
